### PR TITLE
Implement java.util.Collection required methods

### PR DESCRIPTION
**Pull request by michaelrkytch**
I only needed the required (read-only) methods of java.util.Collection for my use case.  Thanks for the pointer to your ordered project.  I also used the implementation of clojure.core.Vec as a reference.

Also, as a relative OSS noob, I'm not sure the etiquette around incrementing the version of someone else's project.

### CODE REVIEW

File: src/amalloy/ring_buffer.clj [hunk 1]
```diff
@@ -47,7 +55,27 @@
       (RingBuffer. start (inc len) (assoc buf (rem (+ start len) (count buf)) x) meta)))
   (seq [this]
     (seq (for [i (range len)]
-           (nth buf (rem (+ start i) (count buf)))))))
+           (nth buf (rem (+ start i) (count buf))))))
+
+  Collection
+  (iterator [this]
+    (SeqIterator. (.seq this)))
```
Comment (amalloy) [inline src/amalloy/ring_buffer.clj]: Don't use SeqIterator, just call `(.iterator ^Iterable (.seq this))`.

File: src/amalloy/ring_buffer.clj [hunk 2]
```diff
@@ -47,7 +55,27 @@
...
+    (every? #(.contains this %) elts))
+  (size [this]
+    (.count this))
+  (isEmpty [this]
+    (empty? this))
+  (toArray [this dest]
+    (reduce (fn [idx item]
+              (aset dest idx item)
+              (inc idx))
+            0, (.seq this))
```
Comment (amalloy) [inline src/amalloy/ring_buffer.clj]: Don't call `.seq` here, just pass `this` and let `reduce` decide whether to seq or do something more efficient instead.

**Comment by amalloy** [comment]
I would suggest not touching the version, in general, unless someone asks you to. Certainly for this library I'd rather you left it alone. It's not a big enough deal that I'd reject the pull request if that were the only issue with it, but since there are some other changes for you to make I'd like you to put the version back to 1.0 while you're at it (and I'll bump it to 1.1 myself for an official release).

**Comment by amalloy** [comment]
Haha I just realized you copied the Collection code from the code I wrote many years ago. Well, that explains it, I guess. I wonder what I was thinking.

**Comment by amalloy** [comment]
Don't worry about the changes; I'll fix them myself, since you were just copying what I told you to copy.

**Comment by michaelrkytch** [comment]
Great.  Thanks for the review!

**Comment by amalloy** [comment]
I just released 1.1-alpha1, so you can try it out and confirm it still does all the stuff you wanted it to after my change, before I release 1.1 final.

**Comment by michaelrkytch** [comment]
Works for me.  Thanks!

**Pull request closed without merging**